### PR TITLE
Initial dev

### PR DESCRIPTION
**Pull request by wistefan**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+[![License badge](https://img.shields.io/github/license/FIWARE/context.Orion-LD.svg)](https://opensource.org/licenses/AGPL-3.0)
```
Comment (jason-fox) [inline README.md:1]: Incorrect link

File: README.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+[![Container Repository on Quay](https://img.shields.io/badge/quay.io-eas-green "Container Repository on Quay")](https://quay.io/repository/wi_stefan/endpoint-configuration-service?tab=tags)
```
Comment (jason-fox) [inline README.md:2]: Will these links need amending after moving to the FIWARE org?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+## APIs
+
+- [Endpoint-Configuration-API](./api/endpoint-configuration-api.yaml)
+- [Auth-Provider-API](./api/auth-provider-api.yaml)
+- [iShare-Credentials-Management-API](./api/ishare-credentials-management-api.yaml)
+
+## Why not use mTLS?
+
+mTLS is a method for mutual authentication. The parties on each end of the connection can be verified through TLS certificates. In contrast to that,
```
Comment (jason-fox) [inline README.md:70]: This should be a link

File: docker-compose/README.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+The components use volumes to share files(envoy-config, certificates) in a very similar manner as it happens inside a kubernetes-sidecar deployment.
+
+## Execution
+
+To start-up the environment run ```docker-compose up``` inside the [docker-compose folder](./). Then execute the [iptables.sh](./iptables.sh) to set up the port-redirection(needs to be run as root).
+
+A request scenario will use the following path(numbers belong to the corresponding one in the diagram):
+
+1. Client sends client credentials to the credentials-management api of the auth-provider: 
+   ```
```
Comment (jason-fox) [inline docker-compose/README.md:46]: Add code highlighting = ```console

File: docker-compose/README.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+        --data-raw '{
+           "certificateChain": "<INSERT_CERTIFICATE>",
+           "signingKey": "<INSERT_SIGNING_KEY>"
+        }'
+   ```
+   The auth provider stores the credentials inside its own filesystem.
+
+
+2. Client send configuration request to the config-service:
+    ```
```
Comment (jason-fox) [inline docker-compose/README.md:58]: Same as previous comment

File: src/endpoint-configuration-service/src/main/java/org/fiware/sidecar/exception/CredentialsConfigNotFound.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
+package org.fiware.sidecar.exception;
+
+import lombok.Getter;
+
+import java.util.List;
+
+public class CredentialsConfigNotFound extends Exception {
```
Comment (jason-fox) [inline src/endpoint-configuration-service/src/main/java/org/fiware/sidecar/exception/CredentialsConfigNotFound.java:10]: Missing JavaDoc describing use of each class.

File: src/endpoint-configuration-service/src/main/java/org/fiware/sidecar/mapping/EndpointMapper.java [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+import org.fiware.sidecar.model.MustacheEndpoint;
+import org.fiware.sidecar.persistence.Endpoint;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.Named;
+
+import java.util.Map;
+import java.util.UUID;
+
+@Mapper(componentModel = "jsr330")
```
Comment (jason-fox) [inline src/endpoint-configuration-service/src/main/java/org/fiware/sidecar/mapping/EndpointMapper.java:17]: Probably worth adding a comment to link to jsr330

File: README.md (hunk 2 above)
Comment (wistefan) [reply README.md:2]: Yes

File: README.md (hunk 1 above)
Comment (wistefan) [reply README.md]: Fixed in ef7698a

File: README.md (hunk 3 above)
Comment (wistefan) [reply README.md]: I rephrased, but what exactly should be a link

File: docker-compose/README.md (hunk 4 above)
Comment (wistefan) [reply docker-compose/README.md]: Fixed in e32e2f1

File: src/endpoint-configuration-service/src/main/java/org/fiware/sidecar/exception/CredentialsConfigNotFound.java (hunk 6 above)
Comment (wistefan) [reply src/endpoint-configuration-service/src/main/java/org/fiware/sidecar/exception/CredentialsConfigNotFound.java:10]: Fixed in a couple of commits

File: src/endpoint-configuration-service/src/main/java/org/fiware/sidecar/mapping/EndpointMapper.java (hunk 7 above)
Comment (wistefan) [reply src/endpoint-configuration-service/src/main/java/org/fiware/sidecar/mapping/EndpointMapper.java:20]: Done

**Pull request merged**